### PR TITLE
move rcedit step to build rather than packaging

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Fix https://github.com/PowerShell/PowerShell/issues/5403

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -558,6 +558,20 @@ Fix steps:
...
     }
 
+    if ($Environment.IsWindows) {
+        ## need RCEdit to modify the binaries embedded resources
+        if (-not (Test-Path "~/.rcedit/rcedit-x64.exe"))
+        {
+            throw "RCEdit is required to modify pwsh.exe resources, please run 'Start-PSBootStrap' to install"
+        }
+
+        $ProductVersion = (git --git-dir="$PSScriptRoot/.git" describe) -Replace '^v'
```
Comment (TravisEz13) [inline build.psm1]: Use release tag parameter to get version if it was passed it.  This is needed for release builds.

Comment (TravisEz13) [inline build.psm1]: Please use `Get-PSCommitId` instead of rewriting this code `git --git-dir="$PSScriptRoot/.git" describe`

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: build.psm1 [hunk 2]
```diff
@@ -565,10 +565,13 @@ Fix steps:
             throw "RCEdit is required to modify pwsh.exe resources, please run 'Start-PSBootStrap' to install"
         }
 
-        $ProductVersion = (git --git-dir="$PSScriptRoot/.git" describe) -Replace '^v'
+        if (!$ReleaseTag)
+        {
+            $ReleaseTag = Get-PSCommitId
```
Comment (TravisEz13) [inline build.psm1]: you still need to remove the v at the beginning.
Get-PSVersion will do that for you, but I don't have a function to do this for the releasetag.

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: build.psm1 [hunk 3]
```diff
@@ -558,6 +558,28 @@ Fix steps:
...
+        ## need RCEdit to modify the binaries embedded resources
+        if (-not (Test-Path "~/.rcedit/rcedit-x64.exe"))
+        {
+            throw "RCEdit is required to modify pwsh.exe resources, please run 'Start-PSBootStrap' to install"
+        }
+
+        $ReleaseVersion = ""
+        if ($ReleaseTag)
+        {
+            $ReleaseVersion = $ReleaseTag -replace '^v'
```
Comment (TravisEz13) [inline build.psm1]: $ReleaseTagToUse should already be this value

**Review by TravisEz13 — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]
LGTM

**Comment by bergmeister** [comment]
Minor but worth pointing out: The bracing style is inconsistent.

**Comment by SteveL-MSFT** [comment]
@bergmeister I think that's a fair point.  Will update for the code I added.

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
The macOS CI was canceled by Travis-CI ...
Given that changes in this PR don't affect PowerShell on Linux/macOS, I will merge this PR.

**Comment by daxian-dbw** [comment]
@TravisEz13 note that even the step is moved to build phase, it's the `publish\pwsh.exe` that gets modified, so you need to copy the `pwsh.exe` from the publish folder when doing the signing.

**Pull request merged**